src/scales/scale.ts

```diff
@@ -37,7 +37,11 @@ module Plottable {
     }
 
     protected _getAllExtents(): D[][] {
-      return d3.merge(this._extentsProviders.values().map((provider) => provider(this)));
+      var providerArray: D[][][] = [];
+      this._extentsProviders.forEach((provider: Scales.ExtentsProvider<D>) => {
+        providerArray.push(provider(this));
+      });
+      return d3.merge(providerArray);
```

Comment:
I looked over the MDN idea for `.values()` and it seems that usage of `.values()` is not too clean?  See if the idea of `values()` / `keys()` is too annoying and if it's too annoying keep as is.